dump/src/util/dump/UniqueIndex.java

```diff
@@ -647,7 +661,7 @@ void update( long pos, E oldItem, E newItem ) {
       if ( noChange ) {
          return;
       }
-      delete(oldItem, pos); // remove from memory
+      delete0(oldItem, pos); // remove from memory
```

Comment:
Why do we call the internal method here without compaction?

Reply:
An update consists of delete+add, so in terms of used size, nothing will change after update completes. If rehash occurs prior to the add part, the incoming item will not be considered in size calculation.

Worst case: a series of delete with a subsequent update, where the delete part crosses the compaction boundary, and the resulting capacity requires another rehash to a larger size to accomodate newItem during the add part. That's why we'd better skip conditional compaction here.

We might explicitly check at the end of the whole update routine without making things worse. Then again, only delete() is going to leave the hash table less occupied than before, which is why we should only check there.